### ISSUE TITLE
Notebook mode - join with saved SQL question

### ISSUE THREAD

**Issue by dfx413** [metabase/metabase#10424]
Hello, 
I've been testing the notebook mode for a while (Branch: notebook-mode
Hash: e7d1c3a) and let me first say that it is really awesome!

I played with the new JOIN feature a little bit and found out that you can select a saved question as a join table. Really cool, nice way to create segments for example.

However, it seems this functionality is either not done yet or not supposed to be there at all. Or maybe I am doing something wrong.

This is how I tried to compose the data:
![image](https://user-images.githubusercontent.com/1764151/62034437-4e917880-b1ee-11e9-82b7-5e84763e4876.png)
 
And here is the resulting query:
```
SELECT TOP 1048576 [LONG LIST OF COLUMNS]
FROM 
-- THIS IS THE REGULAR TABLE
    "dbo"."_v_L4_KL_KLIENT" 
-- THIS IS THE SAVED QUESTION    
LEFT JOIN (SELECT NULL.*) "table_card__334" ON "dbo"."_v_L4_KL_KLIENT"."ID_L4_KL_KLIENT" = "table_card__334"."ID_L4_KL_KLIENT"
-- THESE ARE THE JOINS SPECIFIED IN THE TABLE RELATIONS IN DATA MODEL
LEFT JOIN "dbo"."L4_CIS_STAT" "L4_CIS_STAT__via__ID_L4_CIS_ST" ON "dbo"."_v_L4_KL_KLIENT"."ID_L4_CIS_STAT" = "L4_CIS_STAT__via__ID_L4_CIS_ST"."ID_L4_CIS_STAT" 
LEFT JOIN "dbo"."_vc_POHLAVI_KLIENT" "_vc_POHLAVI_KLIENT__via__POHLA" ON "dbo"."_v_L4_KL_KLIENT"."POHLAVI_KLIENT" = "_vc_POHLAVI_KLIENT__via__POHLA"."ID" 
LEFT JOIN "dbo"."L4_CIS_STAT" "L4_CIS_STAT__via__RODNY_STAT" ON "dbo"."_v_L4_KL_KLIENT"."RODNY_STAT" = "L4_CIS_STAT__via__RODNY_STAT"."ID_L4_CIS_STAT" 
```

I am using MSSQL2016.

Thank you.

**Comment by mazameli** [comment]
What's the behavior when you try to visualize the results of the join? Are you seeing any errors, either on screen or in the javascript or server logs? What is happening that didn't expect? Thanks!

**Comment by dfx413** [comment]
@mazameli Hi, the problem can be seen right in the generated SQL: 
`LEFT JOIN (SELECT NULL.*) "table_card__334"`
which is the part representing the saved question.

**Comment by mazameli** [comment]
Got it. Seems like this is specifically when trying to join to a saved _SQL_ question. Is that what's going on in your case? I've made a note of it here: https://github.com/metabase/metabase/pull/9592#issuecomment-516144809

**Comment by dfx413** [comment]
@mazameli Just to clarify, the saved question I was talking about was created using the query builder, not raw SQL.

**Comment by mazameli** [comment]
Hrmm, okay. Did your saved question have anything interesting about it, like did it include any custom columns, or segments or metrics (defined in the Data Model section), or joins? Was it a question that was created before you tried out this preview build? If so, if you alter something about it and then save it as a different question, and _then_ try joining, does it work then?

**Comment by dfx413** [comment]
@mazameli I am very sorry, you are right, it was an SQL-based question, I got it mixed up.
However, I stumbled upon different type of error with query builder based question.
But it is a little complex to explain so I will follow up later. 
Thank you and again, sorry for my mistake.

**Comment by mazameli** [comment]
No problem! Thanks for clearing that up.

**Comment by boerwastaken** [comment]
I'm trying to do the same - join to a saved SQL question. I'm using v0.33-RC1.
The resulting SQL is of the following form:

```sql
SELECT "mb"."faoca19"."rid" AS "rid", "table_card__76"."rid" AS "rid_2"
FROM "mb"."faoca19"
LEFT JOIN (SELECT NULL.*) "table_card__76" ON "mb"."faoca19"."rid" = "table_card__76"."rid"
```

First of all, there's the `(SELECT NULL.*)` which generates a syntax error (PostgreSQL 12: `ERROR: syntax error at or near "." Position: 233`), upon removing that, of course we see that there's no `table_card__76` to join with: `ERROR: relation "table_card__76" does not exist Position: 221`.


Some background on what I'm trying to do, for which I think I need to use custom SQL:
I have a table A and a table B, with `1:(1..)` relationship. Now, I want to select all *distinct* records from A, for which at least 1 record in B exists that satisfies some condition. The problem is: I can't get to *distinct* records from A using the query builder. I'd like to use "SELECT DISTINCT" but I can't use `DISTINCT` outside some summarization function - I can get a ***count*** of distinct values (one field), but not the distinct values themselves.

**Comment by camsaul** [comment]
This may have been fixed by #10517

**Comment by flamber** [comment]
@camsaul 
Just tested with release-0.33.x (commit 7ab9d9351adff5468b784c8364beafba7ba0bf76)
By creating a SQL question like this (on Sample H2):
```
SELECT 1 AS "ID", 'Extra' AS "Extra"
```
And then creating a notebook by Orders, where I join the above question and gives the `NULL.*` error:
```
Syntax error in SQL statement " SELECT ""PUBLIC"".""ORDERS"".""ID"" AS ""ID"", ""PUBLIC"".""ORDERS"".""CREATED_AT"" AS ""CREATED_AT"", ""PUBLIC"".""ORDERS"".""DISCOUNT"" AS ""DISCOUNT"", ""PUBLIC"".""ORDERS"".""PRODUCT_ID"" AS ""PRODUCT_ID"", ""PUBLIC"".""ORDERS"".""QUANTITY"" AS ""QUANTITY"", ""PUBLIC"".""ORDERS"".""SUBTOTAL"" AS ""SUBTOTAL"", ""PUBLIC"".""ORDERS"".""TAX"" AS ""TAX"", ""PUBLIC"".""ORDERS"".""TOTAL"" AS ""TOTAL"", ""PUBLIC"".""ORDERS"".""USER_ID"" AS ""USER_ID"", ""table_card__19"".""ID"" AS ""ID_2"", ""table_card__19"".""Extra"" AS ""Extra"" FROM ""PUBLIC"".""ORDERS"" LEFT JOIN (SELECT NULL.[*]*) ""table_card__19"" ON ""PUBLIC"".""ORDERS"".""ID"" = ""table_card__19"".""ID"" LIMIT 2000 "; expected "[, ::, *, /, %, +, -, ||, ~, !~, NOT, LIKE, ILIKE, REGEXP, IS, IN, BETWEEN, AND, OR, AS, ,, FROM, WHERE, GROUP, HAVING, UNION, MINUS, EXCEPT, INTERSECT, ORDER, OFFSET, FETCH, LIMIT, FOR, )"; SQL statement: -- Metabase:: userID: 1 queryType: MBQL queryHash: d71f128107d85e95946f195b2f643dbd1c2df37fbbcee41f272ceea7185c1241 SELECT "PUBLIC"."ORDERS"."ID" AS "ID", "PUBLIC"."ORDERS"."CREATED_AT" AS "CREATED_AT", "PUBLIC"."ORDERS"."DISCOUNT" AS "DISCOUNT", "PUBLIC"."ORDERS"."PRODUCT_ID" AS "PRODUCT_ID", "PUBLIC"."ORDERS"."QUANTITY" AS "QUANTITY", "PUBLIC"."ORDERS"."SUBTOTAL" AS "SUBTOTAL", "PUBLIC"."ORDERS"."TAX" AS "TAX", "PUBLIC"."ORDERS"."TOTAL" AS "TOTAL", "PUBLIC"."ORDERS"."USER_ID" AS "USER_ID", "table_card__19"."ID" AS "ID_2", "table_card__19"."Extra" AS "Extra" FROM "PUBLIC"."ORDERS" LEFT JOIN (SELECT NULL.*) "table_card__19" ON "PUBLIC"."ORDERS"."ID" = "table_card__19"."ID" LIMIT 2000 [42001-197]
```

**Comment by camsaul** [comment]
Fixed by #10546
